model/postgres/postgres_server.rb

```diff
@@ -159,6 +182,17 @@ def failover_target
     target[:server]
   end
 
+  def read_replica_failover_target
+    target = resource.servers
```

Comment:
I'd add an `if/else` branch and embed this into `failover_target` method. My reasoning is that callers wouldn't need to pick one or the other function, `failover_target` would abstract away the complexity. Actually, we might not need branching.

Reply:
```ruby
  def failover_target
    target = resource.servers
      .reject { _1.representative_at }
      .select { _1.strand.label == "wait" && !_1.needs_recycling? }
      .map { {server: _1, lsn: _1.run_query("SELECT pg_last_wal_receive_lsn()").chomp} }
      .max_by { lsn2int(_1[:lsn]) }

    return nil if target.nil?

    if resource.ha_type == PostgresResource::HaType::ASYNC
      return nil if lsn_monitor.last_known_lsn.nil?
      return nil if lsn_diff(lsn_monitor.last_known_lsn, target[:lsn]) > 80 * 1024 * 1024 # 80 MB or ~5 WAL files
    end
    target[:server]
  end
```

This seems to cover all cases, but please double check.

Reply:
Yeap, I have updated it as requested